packages/react/src/components/IconButton/index.tsx

```diff
@@ -1,3 +1,10 @@
+/**
+ * Unfortunately, eslint does not recognize the Polymorphic component has propTypes set
+ *
+ * We might be able to remove this if we upgrade eslint and associated plugins
+ * See: https://github.com/dequelabs/cauldron/issues/451
```

Comment:
I started to try upgrading eslint and associated plugins, but we are pretty far out of date. It popped 88 problems that needed to be addressed. I figured it would be better to handle upgrading in a separate PR